lib/immobilienscout/request.rb

```diff
@@ -12,15 +12,16 @@ def initialize(url, params = nil)
 
     def get
       auth_header = generate_auth_header('GET', @params)
-      headers = generate_headers(auth_header)
 
       url_with_query = "#{@url}?#{@params.to_query}"
       uri = URI.parse(url_with_query)
 
       request = Net::HTTP::Get.new(uri)
-      request.initialize_http_header(headers)
+      request.add_field('Authorization', auth_header)
+      request.add_field('Accept', 'application/xml')
 
-      execute_request(uri, request)
+      response = Net::HTTP.start(uri.host, uri.port, use_ssl: true) { |http| http.request request }
+      Immobilienscout::Parsers::Xml.new(response).parse
```

Comment:
Extracting an XML parser is a nice first step, but this refactoring isn't there yet. We still force all GET requests to use the XML version, which is not intended.

Also the HTTP headers are still set here in this class, mixing concerns with the parsing step (both need to match in order to be working, so they should be in one component.

Let's try the strategy pattern here with something like `RequestStrategy`, which includes the code to generate headers and parse the response for either JSON or XML. If you've never used this pattern before, we can also do a pairing session for this if you like.